### DISCUSSION TITLE
Change colors not working for GeoJsonLayer when attribute data is a useState variable.

### DISCUSSION THREAD

**Discussion by even996** [visgl/deck.gl#5915] [Q&A] [answered]
Hi!
I want to change the color of the layer on an event change. For example a onClick button change this color to a other color.
Inside the `GeoJsonLayer `this atribute is `getLineColor `I want to change this color attribute depending on a variable. 
This behavior works when the `data` attribute in `GeoJsonLayer `is a `variable `and it is not working for a `state varaible` like i want.

GeoJsonLayer({
      id: "geojson-layer ",
      data: data, --> data coming from a useState not working. Swap to geojson2 to make it work.

Thanks for help :) 

Code -->  https://codesandbox.io/s/interesting-thompson-e8ntf?file=/src/App.js

**Comment by Pessimistress** [comment, marked as answer]
[updateTriggers](https://deck.gl/docs/api-reference/core/layer#updatetriggers)

**Reply by even996** [reply]
Thanks. 
Do you know if updateTriggers works for nebula's EditableGeoJsonLayer allso?
